Automattic/jetpack
# VideoPress: Fix dashboard fatal mistake when reading malformed local video

Fixes #28707

## Proposed changes:

* Checks if `$media_details` or `$jetpack_videopress` are objects instead of arrays, marking the file as malformed and converting the properties to array, avoiding a `Cannot use object of type stdClass as array` exception.

### Other information:


## Jetpack product discussion


p1676573126334559-slack-C03TA48NSUX

## Does this pull request change what data or activity we track or use?


No.

## Testing instructions:






Testing locally:
* Upload 2 videos to the local library (Media > Add New)

For the first video we will simulate a database problem:
* Get the `post_id` of the video
* Update the database entry in the `wp_postmeta` table for the video's `post_id`, setting `_wp_attachment_metadata` to an empty string (`localhost:8181/` locally)
* ![2023-02-17_12-55-21](https://user-images.githubusercontent.com/8486249/219702469-da92011c-22ae-47f8-b077-cff0c7eb55c8.png)
![2023-02-17_12-55-54](https://user-images.githubusercontent.com/8486249/219702454-5f47587b-0ccc-418e-8ea8-87ffa2dec5e9.png)

For the second video we will remove the file to cause a read error:
* Go to `/wp-content/uploads/<year>/<month>`. Locally and today that would be `tools/docker/wordpress/wp-content/uploads/2023/02`
* Delete or rename the video file: `sudo mv file.mp4 file.mp4.bak`

* Go to the VideoPress dashboard
* Check that there is no fatal error and the dashboard is accessible
* Check that both videos are listed with a warning and cannot be uploaded

Before | After
-|-
![2023-02-16_17-03-25](https://user-images.githubusercontent.com/8486249/219493687-141e5ca1-478e-4880-b8f2-6be167dab869.png) | ![2023-02-16_18-43-53](https://user-images.githubusercontent.com/8486249/219493810-6e67cbdf-8973-4435-9c06-6cf50e1b98c4.png)
